packages/protocol/contracts/team/proving/ProverSet.sol

```diff
@@ -1,63 +1,92 @@
 
     error INVALID_STATUS();
     error PERMISSION_DENIED();
 
+    modifier onlyAuthorized() {
+        if (msg.sender != admin && msg.sender != IHasRecipient(admin).recipient()) {
+            revert PERMISSION_DENIED();
+        }
+        _;
+    }
+
+    modifier onlyProver() {
+        if (!isProver[msg.sender]) revert PERMISSION_DENIED();
+        _;
+    }
+
     /// @notice Initializes the contract.
-    function init(address _owner, address _addressManager) external initializer {
+    function init(
+        address _owner,
+        address _admin,
+        address _addressManager
+    )
+        external
+        nonZeroAddr(_admin)
+        initializer
+    {
         __Essential_init(_owner, _addressManager);
 
-        IERC20 tko = IERC20(resolve(LibStrings.B_TAIKO_TOKEN, false));
+        admin = _admin;
 
-        address taikoL1 = resolve(LibStrings.B_TAIKO, false);
-        tko.approve(taikoL1, type(uint256).max);
-
-        address assignmentHook = resolve(LibStrings.B_ASSIGNMENT_HOOK, false);
-        tko.approve(assignmentHook, type(uint256).max);
+        IERC20 tko = IERC20(resolve(LibStrings.B_TAIKO_TOKEN, false));
+        tko.approve(resolve(LibStrings.B_TAIKO, false), type(uint256).max);
+        tko.approve(resolve(LibStrings.B_ASSIGNMENT_HOOK, false), type(uint256).max);
     }
 
     /// @notice Enables or disables a prover.
-    function enableProver(address _prover, bool _isProver) external onlyOwner {
+    function enableProver(address _prover, bool _isProver) external onlyAuthorized {
         if (isProver[_prover] == _isProver) revert INVALID_STATUS();
         isProver[_prover] = _isProver;
 
         emit ProverEnabled(_prover, _isProver);
     }
 
-    /// @notice Withdraws Taiko tokens back to the owner address.
-    function withdraw(uint256 _amount) external onlyOwner {
-        IERC20(resolve(LibStrings.B_TAIKO_TOKEN, false)).transfer(owner(), _amount);
+    /// @notice Withdraws Taiko tokens back to the admin address.
+    function withdrawToAdmin(uint256 _amount) external onlyAuthorized {
+        IERC20(resolve(LibStrings.B_TAIKO_TOKEN, false)).transfer(admin, _amount);
     }
 
     /// @notice Proves or contests a Taiko block.
-    function proveBlock(uint64 _blockId, bytes calldata _input) external whenNotPaused {
-        if (!isProver[msg.sender]) revert PERMISSION_DENIED();
-
+    function proveBlock(uint64 _blockId, bytes calldata _input) external onlyProver nonReentrant {
         emit BlockProvenBy(msg.sender, _blockId);
         ITaikoL1(resolve(LibStrings.B_TAIKO, false)).proveBlock(_blockId, _input);
     }
 
+    /// @notice Delegates token voting right to a delegatee.
+    /// @param _delegatee The delegatee to receive the voting right.
+    function delegate(address _delegatee) external onlyAuthorized nonReentrant {
+        ERC20VotesUpgradeable(resolve(LibStrings.B_TAIKO_TOKEN, false)).delegate(_delegatee);
+    }
```

Comment:
This one too, shall be put also to `TokenUnlocking` (to allow calling into this function) ?

Reply:
Sorry I don't understand this feedback, why duplicate logics in two contracts?

Reply:
Yep, i missed the modifier change. Seems no issue indeed.